test/functional/node_network_limited.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+from test_framework.mininode import *
+
+class BaseNode(P2PInterface):
+    nServices = 0
+    firstAddrnServices = 0
+    def on_version(self, message):
+        self.nServices = message.nServices
+
+class NodeNetworkLimitedTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+        self.extra_args = [['-prune=550']]
+
+    def getSignaledServiceFlags(self):
+        node = self.nodes[0].add_p2p_connection(BaseNode())
+        NetworkThread().start()
+        node.wait_for_verack()
+        services = node.nServices
+        self.nodes[0].disconnect_p2ps()
+        node.wait_for_disconnect()
+        return services
+
+    def tryGetBlockViaGetData(self, blockhash, result):
+        node = self.nodes[0].add_p2p_connection(BaseNode())
+        NetworkThread().start()
+        node.wait_for_verack()
+        node.send_message(msg_verack())
+        getdata_request = msg_getdata()
+        getdata_request.inv.append(CInv(2, int(blockhash, 16)))
+        node.send_message(getdata_request)
+        # check if we the peer sends us the block
+        res = False
+        try:
+            node.wait_for_block(int(blockhash, 16), 3)
+            res = True
+            assert_equal(res, result)
```

Comment:
It seems we never check anything if we got disconnected, I guess this check needs to move to after the except?